### DISCUSSION TITLE
Can't save quarto docs to html if they have ojs runtime

### DISCUSSION THREAD

**Discussion by smach** [quarto-dev/quarto-cli#7114] [Q&A] [answered]
I used to be able to save quarto html files to a single HTML file even if they included Observable JavaScript. Since I upgraded to the latest version of quarto, I can no longer do this. The files preview OK, but they don't work as a single HTML file. I just tried to see what happens if I save it to multiple files and got this error message:

The OJS runtime does not work with file:// URLs. Please use a web server to view this document.

I've been sharing monthly reports at work as a single HTML file that included Observable code, and it's a problem if it won't work anymore. Is there some way to get this capability back?

**Comment by cscheid** [comment]
@smach Unfortunately, the capability was never there to begin with; if it was working for you, that was honestly very lucky. Those files need to be served from a http:// or https:// server because of security restrictions coming from the OJS runtime.

**Reply by smach** [reply]
I just tried rolling back to quarto 1.1.189 and it works. I guess I'll have one machine that never upgrades! (well, eventually a Docker container)

**Reply by cscheid** [reply, marked as answer]
For what's worth, quarto 1.2 also has an older version of ObservableHQ's runtime that apparently works (partially) with the file:// scheme.

quarto 1.3 already doesn't, and unfortunately it won't in 1.4 and going forward. There's nothing we can do about it.

**Reply by smach** [reply]
Good to know, thank you. I'll test out 1.2 and then set up a Docker container with one of them.
